### ISSUE TITLE
clean up after Revision Refactoring

### ISSUE THREAD

**Issue by splitbrain** [splitbrain/dokuwiki#3576]
It seems that there are a few issues left after #3361 has been merged and I want to collect them here to make sure we don't forget them.

- [x] when opening the new files in IntelliJ Idea, I see quite a lot of warnings. Eg. some variables are undefined, others never used, return types are undefined. #3581
- [x] `PageFile` initializes a `PageChangeLog` on construction. This in turn will read a chunk from the changelog file. I am not sure this is required for every use of PageFile? Can you confirm that every use of PageFile will subsequently need a read from the changelog anyway? If not, this should be initialized on demand only. => Moved loading of current revision info #3581
- [x] ?do=diff for a just created page results in error because for left side no revision is available #3607
- [x] comparing different revision combinations with `rev2[]`, which can be chosen at ?do=revisions with checkboxes, or at ?do=diff via the dropdown menus or forward-backward arrows, result in error if the right hand side is an external deletion revision. This because the external deletion revision has no timestamp associated, so therefore a current time is used. If you chose that combination via checkbox/dropdown/arrows the timestamp is older at the moment of the page reload, and there is no mechanism to trace it back to the external deletion revision. #3607

@ssahara send me a private mail to get one of our free IntelliJ licenses

@Klap-in can you please extend the list?

**Comment by Klap-in** [comment]
First point addressed with #3581

**Comment by Klap-in** [comment]
Second point also addressed in #3581

For 3: @ssahara, I see you are solving the issue with not existing revision in a the diff view https://github.com/splitbrain/dokuwiki/compare/master...ssahara:revisiionHandle3x

For 4: @ssahara, do you also look into support for rev2[] with external deletions? Or shall I try? I think timestamp 9999999 is the easiest way out, but maybe you have other creative ideas?

**Comment by ssahara** [comment]
For 4: @Klap-in , the rev2[] is now validated using `ChangeLog::getRelativeRevision($rev2[1]-1, +1)` to follow moving timestamp for external deleted page. please see https://github.com/splitbrain/dokuwiki/compare/master...ssahara:revisiionHandle3x . If it looks good, I will make PR to #3581 revdiffrefactor.

**Comment by Klap-in** [comment]
Aha, I did not notice it yet, that you solved that already. I will test it as well.

**Comment by ssahara** [comment]
I understand using getRelativeRevision() is less performance. I have updated code after merging changes from revdiffrefactor branch into https://github.com/splitbrain/dokuwiki/compare/master...ssahara:revisiionHandle3x

**Comment by Klap-in** [comment]
The PR #3596 was at the branch that is merged in the meantime. @ssahara could you make a new PR?

**Comment by ssahara** [comment]
@Klap-in , I will create a new PR. for the moment my pc is occupied some calculation of experiment data.

**Comment by Klap-in** [comment]
Implemented by the PR of @ssahara
